src/go/rpk/pkg/config/params.go

```diff
@@ -270,6 +272,60 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 	return c, nil
 }
 
+// Write writes loaded configuration parameters to redpanda.yaml.
+func (c *Config) Write(fs afero.Fs) error {
+	cfgPath := c.loadedPath
+	if cfgPath == "" {
+		cfgPath = Default().ConfigFile
+	}
+
+	b, err := yaml.Marshal(c)
+	if err != nil {
+		return fmt.Errorf("marshal error in loaded config, err: %s", err)
+	}
+
+	// Create a temp file.
+	layout := "20060102150405" // year-month-day-hour-min-sec
+	bFilename := "redpanda-" + time.Now().Format(layout) + ".yaml"
+	temp := filepath.Join(filepath.Dir(cfgPath), bFilename)
+
+	err = afero.WriteFile(fs, temp, b, 0o644) // default permissions 644
+	if err != nil {
+		return fmt.Errorf("error writing to temporary file in %s, err: %s", temp, err)
+	}
+
+	// If we have a loaded file we keep permission and ownership of the
+	// original config file.
+	if c.loadedPath != "" {
+		stat, err := fs.Stat(c.loadedPath)
+		if err != nil {
+			return err
+		}
+
+		err = fs.Chmod(temp, stat.Mode())
+		if err != nil {
+			return err
+		}
+
+		// Stat_t is only valid in unix not on Windows.
+		if stat, ok := stat.Sys().(*syscall.Stat_t); ok {
+			gid := int(stat.Gid)
+			uid := int(stat.Uid)
+			err = fs.Chown(temp, uid, gid)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = fs.Rename(temp, cfgPath)
+	if err != nil {
+		return fmt.Errorf("error writing to config file in %s, saved temp file in %s, err: %s", cfgPath, temp, err)
```

Comment:
`return fmt.Errorf("unable to rename temp file %q to existing file %q: %v", temp, cfgPath, err)`